File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.cpp
class Solution {
   public:
    int findMin(vector<int>& nums) {
        int const numSize = static_cast<int>(nums.size());
        int l = 0;
        int r = numSize - 1;
        int m = (l + r) / 2;

        while (l < r) {
            if (nums[m] < nums[r]) {
                r = m;
                m = (l + r) / 2;
            } else {
                l = m + 1;
                m = (l + r) / 2;
            }
        }

        return nums[l];
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-2.rs
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
let mut num_index_map: HashMap<i32, i32> = HashMap::new();

        for (i, &num) in nums.iter().enumerate() {
            let diff = target - num;

            if let Some(&index) = num_index_map.get(&diff) {
                return vec![index, i as i32];
            }

            num_index_map.insert(num, i as i32);
        }

        vec![]
    }
}
